Add bare metal promscale stack deployment docs (#446)
timescale/promscale: docs/bare-metal-promscale-stack.md

Follow the steps to compile promscale extension from source available on [Promscale extension page](https://github.com/timescale/promscale_extension).

### 2. Deploying Promscale

Download Promscale

```
LATEST_VERSION=$(curl -s https://api.github.com/repos/timescale/promscale/releases/latest | grep "tag_name" | cut -d'"' -f4)
curl -L -o promscale "https://github.com/timescale/promscale/releases/download/${LATEST_VERSION}/promscale_${LATEST_VERSION}_Linux_x86_64"
```

Grant executable permissions to Promscale:

```
chmod +x promscale
```

To deploy Promscale, run the following command:
```
./promscale --db-name <DBNAME> --db-password <DB-Password> --db-ssl-mode allow
```
Note that the flags `db-name` and `db-password` refer to the name and password of your TimescaleDB database from Step 1.

Furthermore, note that the command above is to deploy Promscale with SSL mode as optional. To deploy Promscale with SSL mode enabled, configure your TimescaleDB instance with ssl certificates and set the `--db-ssl-mode` flag to `require`.  Promscale will then force authentication via SSL.

### 3. Deploying Prometheus

> **Note: Replace the versions of Prometheus listed here with the latest available version numbers**

Reviewer: should we use the same latest trick as above instead and get the latest version number into an environmental variable, which we plug into the commands below?

Reviewer 2: I too liked that way.

```
# Add our PPA
sudo add-apt-repository ppa:timescale/timescaledb-ppa
sudo apt-get update

# Now install appropriate package for PG version
sudo apt install timescaledb-postgresql-12
```
#### Configure TimescaleDB
Configure your database using `timescaledb-tune`. This will ensure that TimescaelDB is properly added to the parameter `shared_preload_libraries` as well as offer suggestions for tuning memory, parallelism, and other settings:
```
sudo timescaledb-tune

# Restart PostgreSQL instance
sudo service postgresql restart
```

### Access TimescaleDB

To access TimescaleDB, login to your PostgresSQL database using `psql` as user `postgres`:
```
su - postgres
```

To open a connection to your TimescaleDB instance via psql, run:
```
psql
```

And finally, add TimescaleDB as an extension:
```
CREATE EXTENSION IF NOT EXISTS timescaledb CASCADE;
```

For more instructions on installing TimescaleDB on an operating system other than Ubuntu (e.g Debian, MacOS, CentOS etc) see the [TimescaleDB install page.](https://docs.timescale.com/latest/getting-started/installation)

> 

Reviewer: We are actually missing a step here about making the promscale extension available. We should probably add it as an optional step and just compile from source as described here: https://github.com/timescale/promscale_extension

### 2. Deploying Promscale

Download Promscale

```

> curl -L -o promscale "https://github.com/timescale/promscale/releases/download/0.1.4/promscale_0.1.4_Linux_x86_64"

Reviewer: is there a way for this to point to the latest binary,, not just 0.1.4?

```
# Add our PPA
sudo add-apt-repository ppa:timescale/timescaledb-ppa
sudo apt-get update

# Now install appropriate package for PG version
sudo apt install timescaledb-postgresql-12
```
#### Configure TimescaleDB
Configure your database using `timescaledb-tune`. This will ensure that TimescaelDB is properly added to the parameter `shared_preload_libraries` as well as offer suggestions for tuning memory, parallelism, and other settings:
```
sudo timescaledb-tune

# Restart PostgreSQL instance
sudo service postgresql restart
```

### Access TimescaleDB

To access TimescaleDB, login to your PostgresSQL database using `psql` as user `postgres`:
```
su - postgres
```

To open a connection to your TimescaleDB instance via psql, run:
```
psql
```

And finally, add TimescaleDB as an extension:
```
CREATE EXTENSION IF NOT EXISTS timescaledb CASCADE;
```

For more instructions on installing TimescaleDB on an operating system other than Ubuntu (e.g Debian, MacOS, CentOS etc) see the [TimescaleDB install page.](https://docs.timescale.com/latest/getting-started/installation)

> ### 2. Create a Crontab to enforce retention policies.

Reviewer: are we targeting timescaledb > 2.0, because then this is no longer needed.

```
#/bin/sh
export PGPASSWORD=<PASSWORD>
psql --host localhost --port 5432 --dbname postgres --user postgres -c 'CALL prom_api.execute_maintenance();'
```

Next, create another script, called `install-crontab.sh` and add the code snippet below to it. Ensure to  configure the absolute path to `prom-execute-maintenance.sh`. This script will create a crontab for the retention task:

```
#/bin/sh
echo "0,30 * * * * /<ABSOLUTE_PATH>/prom-execute-maintenance.sh" | crontab -
```

Then, grant executable privileges to both files:

```
chmod +x prom-execute-maintenance.sh
chmod +x install-crontab.sh
```

Finally, to execute the crontab, run:

```
./install-crontab.sh
```

### 3. Deploying Promscale

Download Promscale

```
curl -L -o promscale "https://github.com/timescale/promscale/releases/download/0.1.2/promscale_0.1.2_Linux_x86_64"
```

Grant executable permissions to Promscale:

```
chmod +x promscale
```

To deploy Promscale, run the following command:
```
./promscale --db-name <DBNAME> --db-password <DB-Password> --db-ssl-mode allow
```
Note that the flags `db-name` and `db-password` refer to the name and password of your TimescaleDB database from Step 1.

> Furthermore, note that the command above is to deploy Promscale with SSL mode disabled. To deploy Promscale with SSL mode enabled, configure your TimescaleDB instance with ssl certificates and drop the `--db-ssl-mode` flag.  Promscale will then authenticate via SSL by default.

Reviewer: ```suggestion
Furthermore, note that the command above is to deploy Promscale with SSL mode as optional. To deploy Promscale with SSL mode enabled, configure your TimescaleDB instance with ssl certificates and set the `--db-ssl-mode` flag to `require`.  Promscale will then force authentication via SSL.
```

### 4. Deploying Prometheus

**Note: Replace the versions of Prometheus listed here with the latest available version numbers**

Download prometheus:

```
curl -O -L "https://github.com/prometheus/prometheus/releases/download/v2.22.2/prometheus-2.22.2.linux-amd64.tar.gz"
```

Uncompress the tarball and change directory into the prometheus directory:

```
tar -xzvf prometheus-2.22.2.linux-amd64.tar.gz
```

```
cd prometheus-2.22.2.linux-amd64
```

Edit the `prometheus.yml` file in order to add **Promscale** as a `remote_write` and `remote_read` endpoint for Prometheus: 
 
```
remote_write:
 - url: "http://<promscale-host>:9201/write"
 
remote_read:

>  - url: "http://<promscale-host>:9201/read"

Reviewer: we should probably set read_recent: true here?

  
```

Run Prometheus using the following command: 
```
./prometheus
```

**Note:** Prometheus will load with the configuration defined by the `prometheus.yml` residing in the same directory as the `prometheus` executable.

> ### Deploying Grafana

Reviewer: ```suggestion
### Deploying Grafana (optional)
```
